spec/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source_spec.rb

```diff
@@ -0,0 +1,518 @@
+      let(:expected_result)       { "hello-world(v2.0):example.com/manoj-puthran/sample-scripts/terraform/templates" }
+
+      it "supports https urls" do
+        expect(described_class.template_name_from_git_repo_url(*https_url_branch_path)).to eq(expected_result)
+      end
+
+      it "converts git urls" do
+        expect(described_class.template_name_from_git_repo_url(*git_url_branch_path)).to eq(expected_result)
+      end
+    end
+
+    describe "#update_in_provider" do
+      let(:update_params) { {:scm_branch => "other_branch"} }
+      # let(:notify_update_args) { notification_args('update', update_params) }
+
+      context "with valid params" do
+        it "updates the record and initializes a git repo" do
+          record = build_record
+
+          # expect(Notification).to receive(:create!).with(notify_update_args)
+          # expect(Notification).to receive(:create!).with(notification_args("syncing", {}))
+
+          result = record.update_in_provider update_params
+
+          expect(result).to be_an(described_class)
+          expect(result.scm_branch).to eq("other_branch")
+
+          git_repo_dir = repo_dir.join(result.git_repository.id.to_s)
+          expect(files_in_repository(git_repo_dir)).to eq ["hello_world.tf"]
+
+          names_and_payloads = record.configuration_script_payloads.pluck(:name, :payload)
+
+          names = names_and_payloads.collect(&:first)
+          payloads = names_and_payloads.collect(&:second)
+
+          expect(names.first).to(eq("hello_world_local(other_branch):#{local_repo}/"))
+
+          expected_hash = {
+            "relative_path" => File.dirname(*repo_dir_structure),
+            "files"         => [File.basename(*repo_dir_structure)],
+            "input_vars"    => nil,
+            "output_vars"   => nil
+          }
+
+          expect(payloads.first).to eq(expected_hash.to_json)
+        end
+      end
+
+      context "with invalid params" do
+        it "does not create a record and does not call git" do
+          record                    = build_record
+          update_params[:scm_type]  = 'svn' # oh dear god...
+
+          expect(AwesomeSpawn).to receive(:run!).never
+          # expect(Notification).to receive(:create!).with(notify_update_args)
+
+          expect do
+            record.update_in_provider update_params
+          end.to raise_error(ActiveRecord::RecordInvalid)
+        end
+      end
+
+      context "when there is a network error fetching the repo" do
+        before do
+          record = build_record
+
+          # sync_notification_args        = notification_args("syncing", {})
+
+          # expect(Notification).to receive(:create!).with(notify_update_args)
+          # expect(Notification).to receive(:create!).with(sync_notification_args)
+          expect(record.git_repository).to receive(:update_repo).and_raise(Rugged::NetworkError)
+
+          expect do
+            # described_class.last.update_in_provider update_params
```

Comment:
Same as below... why are these lines commented?  Do we need them?  Let me know if I can help get these tests passing.